### PR TITLE
Split the Frame-related data structures

### PR DESCRIPTION
**Pull request by lu-zero**
This should make easier sharing code with the stand-alone av-scenechange and av-metrics.

- [x] Move Plane
   - [x] Move the math and pixel dependencies
   - [ ] Add a copy_plane function
- [x] Move Frame
   - [x] Make so the exported constructors do not depend on rav1e internals, make the rav1es use them with the internal constant.
- [ ] Move tiling/*

### CODE REVIEW

File: v_frame/Cargo.toml [hunk 1]
```diff
@@ -6,7 +6,13 @@ license = "BSD-2-Clause"
...
 edition = "2018"
 
-
 [dependencies]
 cfg-if = "0.1"
 num-traits = "0.2"
+num-derive = "0.3"
+noop_proc_macro = "0.2.0"
+
+[dependencies.rav1e]
```
Comment (lu-zero) [inline v_frame/Cargo.toml]: It should not depend on rav1e at all.

File: src/frame/mod.rs [hunk 2]
```diff
@@ -41,70 +41,69 @@ pub struct FrameParameters {
...
   #[cfg(test)]
-  pub(crate) fn as_tile(&self) -> Tile<'_, T> {
+  fn as_tile(&self) -> Tile<'_, T> {
     let PlaneConfig { width, height, .. } = self.planes[0].cfg;
     Tile::new(self, TileRect { x: 0, y: 0, width, height })
   }
+}
+
+/// Public Trait for new Mutable Tile of a frame
+pub trait AsTileMut<T: Pixel> {
```
Comment (lu-zero) [inline src/frame/mod.rs]: You may use a single trait for both methods.

File: src/frame/mod.rs [hunk 3]
```diff
@@ -41,70 +41,69 @@ pub struct FrameParameters {
...
+  #[cfg(test)]
+  fn as_tile_mut(&mut self) -> TileMut<'_, T>;
+}
 
+impl<T: Pixel> AsTileMut<T> for Frame<T> {
   #[inline(always)]
   #[cfg(test)]
```
Comment (lu-zero) [inline src/frame/mod.rs]: This way `#[cfg(test)]` can be put before the impl block

File: src/frame/mod.rs [hunk 4]
```diff
@@ -100,9 +100,9 @@ pub trait AsTileMut<T: Pixel> {
   fn as_tile_mut(&mut self) -> TileMut<'_, T>;
 }
 
+#[cfg(test)]
 impl<T: Pixel> AsTileMut<T> for Frame<T> {
```
Comment (lu-zero) [inline src/frame/mod.rs]: AsTileMut should not exist. Have both methods in an AsTile trait.

File: v_frame/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+mod serialize {
+  cfg_if::cfg_if! {
+     if #[cfg(feature="serialize")] {
+       pub use serde::*;
+      } else {
+        pub use noop_proc_macro::{Deserialize, Serialize};
+     }
+  }
+}
+pub mod preamble {
```
Comment (shssoichiro) [inline v_frame/src/lib.rs]: Typically in Rust this is called `prelude`. For consistency, I think it makes sense to use the community-agreed term here.

Comment (lu-zero) [reply v_frame/src/lib.rs]: Agreed, I'm not sure what I thought when I typoed it :)

**Comment by lu-zero** [comment]
The tests need to be fixed.

**Comment by vibhoothiiaanand** [comment]
Let us see now if tests are passing

**Pull request merged**